sync-android/src/test/java/com/cloudant/sync/datastore/encryption/AndroidKeyProviderTests.java

```diff
@@ -0,0 +1,149 @@
+package com.cloudant.sync.datastore.encryption;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import cloudant.com.androidtest.AndroidTestUtil;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class AndroidKeyProviderTests {
+
+    @Test
+    public void testCreateProviderWithIdentifier() {
+        AndroidKeyProvider provider = new AndroidKeyProvider(AndroidTestUtil.context,
+                ProviderTestUtil.password, ProviderTestUtil.getUniqueIdentifier());
+
+        EncryptionKey createdKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not created", createdKey);
+
+        EncryptionKey loadedKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not loaded", loadedKey);
+
+        assertTrue("loadedKey did not match createdKey", Arrays.equals(createdKey.getKey(),
+                loadedKey.getKey()));
+
+        provider.getManager().clearKey();
+    }
+
+    @Test
+    public void testCreateProviderWithManager() {
+        KeyStorage storage = new KeyStorage(AndroidTestUtil.context, ProviderTestUtil
+                .getUniqueIdentifier());
+        KeyManager manager = new KeyManager(storage);
+        AndroidKeyProvider provider = new AndroidKeyProvider(ProviderTestUtil.password, manager);
+
+        EncryptionKey createdKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not created", createdKey);
+
+        EncryptionKey loadedKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not loaded", loadedKey);
```

Comment:
I actually removed that test case as it used KeyManager for creating the AndroidKeyProvider.  The KeyManager class is not public API.  It has package level scope.  I removed the AndroidKeyProvider constructor that takes a KeyManager.  See commit https://github.com/cloudant/sync-android/commit/a93b7fac5ddf8e596ba2df7108476a5b5b18e2c0